src/libprojectM/projectM.hpp

```diff
@@ -407,7 +407,7 @@ class projectM
 #if USE_THREADS
     void ThreadWorker();
 
-    mutable std::mutex m_presetSwitchMutex; //!< Mutex for locking preset switching while rendering and vice versa.
+    mutable std::recursive_mutex m_presetSwitchMutex; //!< Mutex for locking preset switching while rendering and vice versa.
```

Comment:
That'd otherwise cause a deadlock, as both functions are also called in the same stack and thread. I think this lock is just there to prevent/postpone preset switching while rendering a frame at the same time.